src/wallet/wallet.cpp

```diff
@@ -1408,13 +1407,25 @@ bool CWallet::IsHDEnabled()
     return !hdChain.masterKeyID.IsNull();
 }
 
+void CWallet::SetWalletFlag(uint64_t flag)
+{
+    walletFlags |= flag;
```

Comment:
Kind of confusing....
I thought this would be equivalent to `CWallet::SetWalletFlags(uint64_t flags, false)`.

 However  while `CWallet::SetWalletFlags(uint64_t flags, bool memonly)` set the flags to the indicated value, `CWallet::SetWalletFlag(uint64_t flag)` **appends** to it.

Maybe you should rename to `AppendWalletFlags`.